stdlib/2/typing.pyi

```diff
@@ -61,47 +62,60 @@ _V_co = TypeVar('_V_co', covariant=True)  # Any type covariant containers.
 _KT_co = TypeVar('_KT_co', covariant=True)  # Key type covariant containers.
 _VT_co = TypeVar('_VT_co', covariant=True)  # Value type covariant containers.
 _T_contra = TypeVar('_T_contra', contravariant=True)  # Ditto contravariant.
+_TC = TypeVar('_TC', bound=Type[object])
 
-class SupportsInt(metaclass=ABCMeta):
+def runtime(cls: _TC) -> _TC: ...
```

Comment:
IIUC `@runtime` is not defined (yet) in typing.py but (only) in typing_extensions.py?

Reply:
Yes, and the same for ``Protocol`` (there is only old internal ``_Protocol`` in ``typing``). We however need them in typeshed to avoid circular dependencies with ``typing_extensions``. We discussed this with @JukkaL few moths ago, the plan (which I like) is this:
>1. Iterate on PR #3132 until it's approved.
>2. Publish ``Protocol`` in ``typing_extensions``. Also include it in the stub for ``typing_extensions``.
>3. Merge PR #3132 but don't merge typeshed changes yet. This lets mypy users (at least those who use GitHub master) experiment with user-defined protocols.
>4. Get an okay from PyCharm and Pytype that they are fine with merging the typeshed PR. They might need to implement some support for the ``Protocol`` special form first, but if they don't use the ``typing`` typeshed stub this may be a non-issue.
>5. If major issues are found in the mypy implementation of protocols, fix them.
>6. Release a new mypy version with protocols as an experimental feature!
>7. Merge the typeshed PR and use it in mypy.
>8. Seek acceptance of the protocols PEP once we have some practical experience and protocols seem to work fine.
>9. Officially support using protocols in typeshed once Pytype and PyCharm are okay with it.
>10. Release Protocol in Python stdlib typing module.

This PR is step number 7